gnome-initial-setup/eos/hack-sound-client/eos-hack-sound-client.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2019 Endless Mobile. Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *     Cesar Fabian Orccon Chipana <cfoch.fabian@gmail.com>
+ */
+
+#include "eos-hack-sound-client.h"
+
+static void eos_hack_sound_client_initable_iface_init (GInitableIface *iface);
+static void eos_hack_sound_client_finalize            (GObject *object);
+static gboolean eos_hack_sound_client_initable_init (GInitable    *initable,
+                                                     GCancellable *cancellable,
+                                                     GError      **error);
+static void play_sound_cb (GObject      *source_object,
+                           GAsyncResult *res,
+                           gpointer     data);
+static void stop_sound_cb (GObject      *source_object,
+                           GAsyncResult *res,
+                           gpointer     data);
+
+struct _EOSHackSoundClient
+{
+  GObject parent_instance;
+  GDBusProxy *proxy;
+};
+
+#define eos_hack_sound_client_parent_class parent_class
+
+G_DEFINE_TYPE_WITH_CODE(EOSHackSoundClient,
+                        eos_hack_sound_client,
+                        G_TYPE_OBJECT,
+                        G_IMPLEMENT_INTERFACE (G_TYPE_INITABLE,
+                                               eos_hack_sound_client_initable_iface_init);)
+
+static void
+eos_hack_sound_client_class_init (EOSHackSoundClientClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+  object_class->finalize = eos_hack_sound_client_finalize;
+}
+
+static void
+eos_hack_sound_client_init (EOSHackSoundClient *client)
+{
+  /* empty */
+}
+
+static void
+eos_hack_sound_client_finalize (GObject *object)
+{
+  EOSHackSoundClient *client = EOS_HACK_SOUND_CLIENT (object);
+
+  g_object_unref (client->proxy);
```

Comment:
My suggestion was `g_clear_object (&client->proxy);`. Using `g_object_unref()` on a possibly-NULL pointer is an error. This will CRITICAL if `client->proxy` is `NULL`, which it will be if `eos_hack_sound_client_initable_init` fails.

```suggestion
  g_clear_object (&client->proxy);
```